src/NetEscapades.AspNetCore.SecurityHeaders.TagHelpers/AttributeHashTagHelper.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.Linq;
+using System.Text;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.AspNetCore.Mvc.ViewFeatures;
+using Microsoft.AspNetCore.Razor.TagHelpers;
+using NetEscapades.AspNetCore.SecurityHeaders.Infrastructure;
+
+namespace NetEscapades.AspNetCore.SecurityHeaders.TagHelpers;
+
+/// <summary>
+/// Generates an hash of style, or inline script attributes
+/// </summary>
+[HtmlTargetElement("*", Attributes = AttributePrefix + "*")]
+public class AttributeHashTagHelper : TagHelper
+{
+    private const string AttributePrefix = "asp-add-";
+    private const string AttributeSuffix = "-to-csp";
```

Comment:
The prefix helper is obviously very geared towards a fixed prefix, rather than a fixed prefix _and_ suffix 🤔 

I wonder if it makes sense to switch to just a prefix, even if if doesn't read quite as nicely? 🤔 

e.g. 

- `asp-add-csp-for-style`
- `asp-add-csp-for-script`

Reply:
Very good suggestion, totally agree, and applied!

TBH, at some point, it crossed my mind, but I didn't want to break the style from the `asp-add-content-to-csp` for the content tag helper. However, now that I look at it with a fresh mind, this definitely makes more sense.